### ISSUE TITLE
Output definitions in Cuckoofile.toml are ignored/in conflict using CuckooPluginFile.

### ISSUE THREAD

**Issue by JavaAdam** [Brightify/Cuckoo#494]
CuckooPluginFile ist overriding all output definitions by setting CUCKOO_OVERRIDE_OUTPUT to an output on the derived data directory. This leads to two unexpected behaviors:

- When a global output is defined in the Cuckoofile.toml an error is thrown because "two global output definitions" are set.
- Output definitions in modules are ignored at all.

So the Swift Package Manager approach is usable for very simple projects only.

**Comment by MatyasKriz** [comment]
Hey @JavaAdam, thanks for letting me know.

I won't have access to my computer for a few more days so I can't really think of a better solution, do you have one in mind for complex projects?

**Comment by JavaAdam** [comment]
To be honest, i am not familiar with restrictions concerning the Swift Package Manager Plugin. But I would just expect, that the output definitions in Cuckoofile.toml are respected/effective. So that the developer is free to define whatever is needed for his particular project.

When it's not possible to write somewhere else than in the derived data directory, than the developer would be responsible to define correct outputs in derived data. A descriptive message in the build log would be helpful here.

**Comment by dr-star** [comment]
Hello @JavaAdam could you share your Cuckoofile.toml?

> * When a global output is defined in the Cuckoofile.toml an error is thrown because "two global output definitions" are set.

**Comment by JavaAdam** [comment]
Hello @dr-star. Unfortunately, I dropped the branch a while ago.

But as I remember I did something like this:

`[modules.MyProject]
output = "${PROJECT_DIR}/MyTarget/Tests/MyGeneratedMocks.swift"
`

Defining additional a global output failed with a "two global output definitions" error message.

**Comment by soucolline** [comment]
After trying to migrate to 2.0.9 i'm getting the exact same behaviour.
- Setting an output to a specific folder inside a module doesn't work and creates the mock inside the derived data
- Specifying a global output on line 1 of .toml file result in error ```Multiple global `output` parameters. Behavior is undefined```
- Only specifying a global output does the same error

**Comment by GabrielBrodersen** [comment]
Same issue, which results in an error `Filename "GeneratedMocks.swift used twice: ...` although the two outputs I specified in the `Cuckoofile.toml` in my project's root (as per the Installation Instructions) are `output = "ModuleATests/Resources/Mocks.swift"` and `output = "ModuleBTests/Resources/Mocks.swift"`. So I guess the `GeneratedMocks.swift` are some default output location because the `output` property doesn't work.
It appears that although the Installation Instructions indicate how to set up rules for projects with multiple test modules, looking at the plugin code, the `FIXME` could indicate otherwise:
<img width="880" alt="Image" src="https://github.com/user-attachments/assets/d636682a-c0ff-4f8a-843f-41505d986df5" />

**Comment by MatyasKriz** [comment]
Yeah, it's an unfortunate limitation of the SPM build tool, perhaps the duplicate GeneratedMocks.swift could be fixed by instead of using static `GeneratedMocks.swift` file we could pass a UUID like `GeneratedMocks-UUID.swift`.

**Comment by adelamarre-pelm** [comment]
Is there a temporary solution for this issue?
